joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/KtPsiToAst.scala

```diff
@@ -858,53 +976,47 @@ trait KtPsiToAst {
     }
-    val (fullName, signature) =
-      typeInfoProvider.fullNameWithSignature(expr, (astDerivedMethodFullName, astDerivedSignature))
-    registerType(typeInfoProvider.containingDeclType(expr, TypeConstants.any))
-    val retType = registerType(typeInfoProvider.expressionType(expr, TypeConstants.any))
-    val methodName = if (isFieldAccessCall || fullName == Operators.fieldAccess) {
-      Operators.fieldAccess
-    } else {
-      expr.getSelectorExpression.getFirstChild.getText
+    val hasNameRefSelector = expr.getSelectorExpression.isInstanceOf[KtNameReferenceExpression]
+    val isFieldAccessCall  = hasThisSuperOrNameRefReceiver && hasNameRefSelector
+    val isCallToSuper = expr.getReceiverExpression match {
+      case _: KtSuperExpression => true
+      case _                    => false
     }
-    val dispatchType = if (isFieldAccessCall) {
-      DispatchTypes.STATIC_DISPATCH
-    } else if (callKind == CallKinds.DynamicCall) {
-      DispatchTypes.DYNAMIC_DISPATCH
-    } else if (callKind == CallKinds.ExtensionCall) {
-      DispatchTypes.STATIC_DISPATCH
-    } else {
-      DispatchTypes.STATIC_DISPATCH
+    val isStaticMethodCall = typeInfoProvider.isStaticMethodCall(expr)
+    val hasRefToClassReceiver = expr.getReceiverExpression match {
+      case r: KtNameReferenceExpression =>
+        typeInfoProvider.isReferenceToClass(r)
+      case _ =>
+        false
     }
-
-    val _callNode =
-      callNode(expr.getText, methodName, fullName, signature, retType, dispatchType, line(expr), column(expr))
-    val root         = Ast(withArgumentIndex(_callNode, argIdx))
-    val receiverNode = receiverAst.root.get
-    if (isExtensionCall || isCallToSuper) {
-      root
-        .withChild(receiverAst)
-        .withArgEdge(_callNode, receiverNode)
-        .withChildren(argAsts)
-        .withArgEdges(_callNode, argAsts.map(_.root.get))
+    val noAstForReceiver = isStaticMethodCall && hasRefToClassReceiver
+    if (isFieldAccessCall) {
+      astForQualifiedExpressionFieldAccess(expr, argIdx)
+    } else if (isExtensionCall) {
+      astForQualifiedExpressionExtensionCall(expr, argIdx)
+    } else if (isCallToSuper) {
+      astForQualifiedExpressionCallToSuper(expr, argIdx)
     } else if (noAstForReceiver) {
-      root.withChild(receiverAst).withChildren(argAsts).withArgEdges(_callNode, argAsts.map(_.root.get))
+      astForQualifiedExpressionWithNoAstForReceiver(expr, argIdx)
     } else {
-      val ast = root
-        .withChild(receiverAst)
-        .withArgEdge(_callNode, receiverNode)
-        .withChildren(argAsts)
-        .withArgEdges(_callNode, argAsts.map(_.root.get))
-      if (argAsts.size == 1 && argAsts.head.root.get.isInstanceOf[NewMethodRef]) {
-        ast.withReceiverEdge(_callNode, argAsts.head.root.get)
-      } else {
-        ast.withReceiverEdge(_callNode, receiverNode)
-      }
+      astForQualifiedExpressionWithReceiverEdge(expr, callKind, argIdx)
     }
```

Comment:
From the most recent discussion on when to use braces and when not to, this may be a situation for removing them.